### DISCUSSION TITLE
What is the runtime v6.0.0 uses?

### DISCUSSION THREAD

**Discussion by kstingel** [leokhoa/laragon#698] [Q&A]
Seems silly to ask, but is Laragon v6.0.0 built as 32-bit, or 64-bit?

reason I ask is because Taskman & Procman are showing it as a 32 bit application 
given that most modern PCs are now 64-bit, would it be worth making v7.0.0(?) available as a 64-bit app?

within this repo, do we have access to the source code for the executable?
if not, @leokhoa, is there a private repo you would be prepared to share(read access to) 
so we can create a fork and help you work on a 64-bit iteration?

possibly should be in it's own thread, but I'll mention here since it's related ... 
could we have a `'future-GUI'` branch (under `Code` and `Discussion`) please?
this would give those of us who are willing and able to help modernise the look-and-feel 
somewhere to share ideas and provide mockups without impacting on `'master'` branch

**Comment by Pen-y-Fan** [comment]
@leokhoa has been asked for the source code of Laragon, but declined on security grounds. The exe isn't a Windows .NET app, so others may have problems contributing anyway.

I don't think the runtime of the laragon.exe matters, it manages the runtimes of PHP, Apache, MySQL etc, it doesn't need to handle large 64-bit numbers, for example.

**Reply by Pen-y-Fan** [reply]
Open source asked here, for example, https://github.com/leokhoa/laragon/issues/330

**Comment by kstingel** [comment]
I was thinking more for the GUI tweaks
wasn't requesting direct source code per se, just an API of sorts so I knew what to patch into for certain functionality
mainly I was enquiring on 32/64-bit possibility and language to write in

I can attempt to code in Python for any new elements though, and @leokhoa can either merge them in or not

**Comment by kstingel** [comment]
for backend apps (`phpMyAdmin`, `cmder`, `ngrok`, `etc`), should we be using 32/64-bit versions when there is an option?
as at 2024, not all included components maintain 32-bit versions

**Reply by Pen-y-Fan** [reply]
The portable version of Laragon uses 32-bit, as they tend to be smaller. The full version is 64-bit.

See https://laragon.org/download/

**Reply by Pen-y-Fan** [reply]
Also see: https://github.com/leokhoa/laragon/releases
